### PR TITLE
Prohibit dropping the default retention policy

### PR DESCRIPTION
**Pull request by otoolep**
This is to prevents users from putting their system into an awkward
state. It is a policy that all databases must have at least a default
retention policy.

Fixes issue #3699.

### CODE REVIEW

**Comment by otoolep** [comment]
@dgnorton

File: meta/data.go [hunk 1]
```diff
@@ -172,6 +172,11 @@ func (data *Data) DropRetentionPolicy(database, name string) error {
 		return ErrDatabaseNotFound
 	}
 
+	// Prohibit dropping the default retention policy.
+	if di.DefaultRetentionPolicy == name {
```
Comment (corylanou) [inline meta/data.go]: would this ever be case sensitive?

**Comment by corylanou** [comment]
One question, otherwise +1

**Comment by dgnorton** [comment]
+1

File: meta/data.go (hunk 1 above)
Comment (otoolep) [inline meta/data.go]: Retention policy names are already case sensitive, so this just continues with this policy. If we ever change that, we'll need to change a lot more code.

**Pull request merged**